File: pueblo-golf-tournament-api/Dto/HomeClubDto.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Dto
{
    public class HomeClubDto : BaseDto
    {
        public string? Name { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? Province { get; set; }
        public string? Country { get; set; }
    }
    public class HomeClubRawDto
    {
        public string? Name { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? Province { get; set; }
        public string? Country { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/api/dtos/home-club-dto.dart
import 'package:json_annotation/json_annotation.dart';

import '../../models/base-model.dart';

part 'home-club-dto.g.dart';

@JsonSerializable()
class HomeClubDto extends BaseModel {
  final String name;
  final String? address;
  final String? city;
  final String? province;
  final String? country;

  HomeClubDto(this.address, this.city, this.province, this.country,
      {required this.name, required super.id});

  factory HomeClubDto.fromJson(Map<String, dynamic> json) =>
      _$HomeClubDtoFromJson(json);

  Map<String, dynamic> toJson() => _$HomeClubDtoToJson(this);
}
